homeassistant/components/nuki/__init__.py

```diff
@@ -60,11 +60,12 @@ async def async_setup(hass, config):
             continue
 
         for conf in confs:
-            hass.async_create_task(
-                hass.config_entries.flow.async_init(
-                    DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+            if "host" in conf and "port" in conf and "token" in conf:
+                hass.async_create_task(
+                    hass.config_entries.flow.async_init(
+                        DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+                    )
```

Comment:
Lets import `CONF_PORT`, `CONF_HOST`, `CONF_TOKEN` & `DEFAULT_PORT` and handle it like this:

```suggestion
            hass.async_create_task(
                hass.config_entries.flow.async_init(
                    DOMAIN,
                    context={"source": SOURCE_IMPORT},
                    data={
                        CONF_HOST: conf[CONF_HOST],
                        CONF_PORT: conf.get(CONF_PORT, DEFAULT_PORT),
                        CONF_TOKEN: conf[CONF_TOKEN],
                    },
```

And let's make sure we include it in the 2021.5 release

Reply:
This is smart, but don't solve the main-problem I am trying to solve. I created the issue https://github.com/home-assistant/core/issues/48358 because the integration is trying to load the configuration of my template-lock from the configuration.yaml. So there is no key "host" in conf and throws an exception.

My template-lock configuration:
```yaml
lock:
  - platform: template
    name: Building
    value_template: "{{ true }}"
    lock:
      service: switch.turn_off
      data:
        entity_id: switch.door_building
    unlock:
      service: switch.turn_on
      data:
        entity_id: switch.door_building
```

Reply:
@darkson95 are you sure, that it tries to import that configuration? Since the error you posted in your issue is the same as others reported who don't have a seperate lock configuration.

If you have time please test out the suggested change from @frenck and report back if that worked for you. (You maybe need to "dismiss" the discovered configurations by pressing configure -> cancle)

Reply:
@frenck You are right! I added the check to the platform instead of checking if the keys are present.
@pree I am relatively sure that it loads more configurations and tries to initialize them than Nuki. After I installed the check on the platform, I had no more problems. I have included Frenck's suggestion in my commit.